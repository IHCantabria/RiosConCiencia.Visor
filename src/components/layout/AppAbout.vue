<template>
  <div class="app-about">
    <vs-dropdown
      title="About"
      class="toggler"
      vs-trigger-click
      vs-custom-content
    >
      <div class="toggle-icon">
        <object
          :data="require('@/assets/about.svg')"
          type="image/svg+xml"
          style="pointer-events: none;"
        />
      </div>
      <vs-dropdown-menu>
        <div class="about-container">
          <label class="label-version">Version {{ version }}</label>
        </div>
      </vs-dropdown-menu>
    </vs-dropdown>
  </div>
</template>
<script>
export default {
  data() {
    return {
      version: process.env.VERSION
    };
  }
};
</script>
<style lang="scss" scoped>
.toggler {
  &:hover {
    cursor: pointer;
  }
}
.toggle-icon {
  width: 26px;
  padding: 5px;
  margin: 0 4px 0 0;
  &:hover {
    cursor: pointer;
  }
}
.label-version {
  margin: 2px;
}
.about-container {
  justify-content: space-between;
  display: flex;
  flex-flow: nowrap row;
}
</style>
