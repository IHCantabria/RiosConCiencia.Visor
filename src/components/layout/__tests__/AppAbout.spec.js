import { shallowMount, createLocalVue } from "@vue/test-utils";
import AppAbout from "../AppAbout.vue";
import merge from "lodash.merge";
import Vuesax from "vuesax";

const localVue = createLocalVue();
localVue.use(Vuesax);

function createWrapper(overrides) {
  const defaultMountingOptions = {
    propsData: { version: process.env.VERSION },
    localVue
  };
  return shallowMount(AppAbout, merge(defaultMountingOptions, overrides));
}

describe("appAbout.vue", () => {
  //SNAPSHOT TESTS
  it("renders correctly", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
});
