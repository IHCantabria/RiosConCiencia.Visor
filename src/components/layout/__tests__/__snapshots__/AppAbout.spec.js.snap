// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appAbout.vue renders correctly 1`] = `
<div
  class="app-about"
>
  <vs-dropdown-stub
    class="toggler"
    color="primary"
    title="About"
    vscustomcontent="true"
    vstriggerclick="true"
  >
    <div
      class="toggle-icon"
    >
      <object
        data=""
        style="pointer-events: none;"
        type="image/svg+xml"
      />
    </div>
     
    <vs-dropdown-menu-stub>
      <div
        class="about-container"
      >
        <label
          class="label-version"
        >
          Version 
        </label>
      </div>
    </vs-dropdown-menu-stub>
  </vs-dropdown-stub>
</div>
`;
